h/static/scripts/annotation-ui.js

```diff
@@ -375,8 +424,12 @@ module.exports = function ($rootScope, settings) {
 
     /** Add annotations to the currently displayed set. */
     addAnnotations: function (annotations) {
+      var added = annotations.filter(function (annot) {
```

Comment:
Hmm. This seems like a bit of a hack, and it's another `O(N * M)` operation -- so we're now doing two of them. Couldn't we move a bunch of the work being done in the switch above into this function so we don't need to compute this set twice?